_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,596 @@
+that this value was updated but here is very good example where comes `race condition` into play. This race condition
+can occur if the assertion (verify phase) is too quick and value has not been updated. That's why we have these dynamic
+wait. The logic behind this method is that every `pollIntervalMs` for specific period of time `timeoutMs` the method checks
+`BooleanSupplier` condition which user can provide and if this condition is true the method will successfully end. Otherwise, we will get the
+`WaitException` with some reasonable description. In case of interest method is located [here](https://github.com/strimzi/strimzi-kafka-operator/blob/master/test/src/main/java/io/strimzi/test/TestUtils.java#L115-L145).
+
+#### Constants
+
+Purpose of this class is to keep global constants used across system tests. You can find the constants for timeout
+interval, poll internal used inside `waitFor` method. Moreover, you can find timeout for cluster operator and clients.
+Also, we have here specific ports such as KafkaBridge metric, TopicOperator metric, Keycloak and so on. Lastly, here we have all tags names,
+which we used in whole `system test` module for instance `ACCEPTANCE`, `REGRESSION` and more.
+ 
+#### Environment
+
+System tests can be configured by several environment variables, which are loaded before test execution.
+
+Variables can be defined via environmental variables or a configuration file, this file can be located anywhere on the file system as long as a path is provided to this file.
+The path is defined by environmental variable `ST_CONFIG_PATH`, if the `ST_CONFIG_PATH` environmental variable is not defined, the default config file location is used `systemtest/config.json`.
+Loading of system configuration has the following priority order:
+1. Environment variable
+2. Variable defined in configuration file
+3. Default value
+
+Here is the list of important environment variables:
+
+1. `DOCKER_ORG` = Specify the organization/repo containing the image used in system tests (default value `strimzi`)
+2. `DOCKER_TAG` = Specify the image tags used in system tests (default value `latest`)
+3. `SKIP_TEARDOWN` = Variable for skip teardown phase for more debug if needed (default value `false`)
+4. `ST_KAFKA_VERSION` =  Kafka version used in images during the system tests (default value `2.6.0`)
+
+#### Our Kafka clients
+
+Another helpful part of the system tests are Kafka clients. In this module we have three types:
+
+1. Internal using test-client image
+2. Internal using client-examples image
+3. External using Vert.x client
+
+##### Internal using test-client image
+
+The first client using the our image. Basically, it is defined as a deployment and inside the container, we start the 
+`kafka-verifiable-(consumer|producer).sh` shell script. In case of more interest you can [here](https://github.com/strimzi/strimzi-kafka-operator/tree/master/docker-images/test-client).
```

Comment:
```suggestion
`kafka-verifiable-(consumer|producer).sh` shell script. In case of more interest you can [here](https://github.com/strimzi/strimzi-kafka-operator/tree/release-0.20.x/docker-images/test-client).
```